src/controllers/expensesController.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+const expensesService = require('../services/expensesService');
+const usersService = require('../services/usersService');
+const {
+  BAD_REQUEST,
+  NOT_FOUND,
+  CREATED,
+  SUCCESS,
+  NO_CONTENT,
+} = require('../utils/statusCodes');
+
+const timestampRegex = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z$/;
+
+const isPostData = ({
+  userId,
+  spentAt,
+  title,
+  amount,
+  category,
+  note,
+}) => {
+  return typeof userId !== 'number'
+    || typeof spentAt !== 'string'
+    || typeof title !== 'string'
+    || typeof amount !== 'number'
+    || typeof category !== 'string'
+    || typeof note !== 'string'
+    || !usersService.getById(userId)
+    || !timestampRegex.test(spentAt);
+};
+
+const getAll = async(req, res) => {
+  res.status(SUCCESS).json(expensesService.getAll(req.query));
+};
+
+const post = async(req, res) => {
+  if (isPostData(req.body)) {
+    res.status(BAD_REQUEST).end();
+
+    return;
+  }
+
+  res.status(CREATED).json(expensesService.addExpense(req.body));
```

Comment:
better to move this `expensesService.addExpense(req.body)` to variable, it looks to complicated

Reply:
The same in other similar places